examples/hello/bpBZip2Wrapper/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+#------------------------------------------------------------------------------#
+# Distributed under the OSI-approved Apache License, Version 2.0.  See
+# accompanying file Copyright.txt for details.
+#------------------------------------------------------------------------------#
+
+if(ADIOS2_HAVE_MPI)
+  find_package(MPI COMPONENTS C REQUIRED)
+
+  add_executable(hello_bpBZip2Wrapper helloBPBZip2Wrapper.cpp)
+  target_include_directories(hello_bpBZip2Wrapper PRIVATE ${MPI_C_INCLUDE_PATH})
+  target_link_libraries(hello_bpBZip2Wrapper ${MPI_C_LIBRARIES})
+  
+else()
+  add_executable(hello_bpBZip2Wrapper helloBPBZip2Wrapper_nompi.cpp)
+  
+endif()
+
+target_link_libraries(hello_bpBZip2Wrapper adios2 ${BZIP2_LIBRARIES})
```

Comment:
bzip2 is a private dependency of ADIOS.  Since it's not directly being used then the libraries shouldn't be included as a dependency.  Also, if you were to use them (which you shouldn't need, they need to be dropped) then you'd want to use the imported targets instead of the libraries directly, il.e use `BZip2::BZip2` instead of `${BZIP2_LIBRARIES}`